### PR TITLE
Add new `ADMIN_SCANNERS_*` permissions

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/13239

---

This patch adds news permissions for accessing the scanners admin. Because admins have some actions to triage the scanner results, some code was added to handle permissions for those actions.

### CODE REVIEW

File: src/olympia/scanners/admin.py [hunk 1]
```diff
@@ -178,6 +179,20 @@ def has_delete_permission(self, request, obj=None):
     def has_change_permission(self, request, obj=None):
         return False
 
+    def get_list_display(self, request):
+        fields = super().get_list_display(request)
+        return self._excludes_admin_fields(request=request, fields=fields)
+
+    def get_fields(self, request, obj=None):
```
Comment (bobsilverberg) [inline src/olympia/scanners/admin.py]: I see tests for `test_list_view`, but I'm not sure if those are testing `get_fields` or `get_list_display` (or both, although that seems unlikely). I guess my question is: do we have added coverage for both of these methods?

File: src/olympia/scanners/tests/test_admin.py [hunk 2]
```diff
@@ -430,13 +453,52 @@ def test_handle_revert_report(self):
...
+        assert response.status_code == 404
+
 
 class TestScannerRuleAdmin(TestCase):
     def setUp(self):
         super().setUp()
 
         self.user = user_factory()
-        self.grant_permission(self.user, 'Admin:Advanced')
+        self.grant_permission(self.user, 'Admin:*')
```
Comment (bobsilverberg) [inline src/olympia/scanners/tests/test_admin.py]: As above, why this change?

File: src/olympia/scanners/admin.py [hunk 3]
```diff
@@ -178,6 +179,20 @@ def has_delete_permission(self, request, obj=None):
...
+        fields = super().get_fields(request, obj)
+        return self._excludes_admin_fields(request=request, fields=fields)
+
+    def _excludes_admin_fields(self, request, fields):
+        is_admin = acl.action_allowed(request, amo.permissions.ADMIN_ADVANCED)
+        if not is_admin:
+            return list(filter(lambda x: x != 'result_actions', fields))
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/scanners/admin.py]: It is indeed only used to exclude this field (for now). And yes, this field contains admin links.

File: src/olympia/scanners/admin.py [hunk 4]
```diff
@@ -239,7 +254,8 @@ def formatted_matched_rules(self, obj):
     formatted_matched_rules.short_description = 'Matched rules'
 
     def handle_true_positive(self, request, pk, *args, **kwargs):
-        if request.method != "POST":
+        is_admin = acl.action_allowed(request, amo.permissions.ADMIN_ADVANCED)
+        if not is_admin or request.method != "POST":
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/scanners/admin.py]: There was no check previously because only admins could get to this page (behind a VPN). Adding a new permission made me think that now would be a good time to actually secure those actions, too.

File: src/olympia/scanners/tests/test_admin.py [hunk 5]
```diff
@@ -39,7 +39,7 @@ def setUp(self):
...
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/scanners/tests/test_admin.py]: That's because most test cases assume the current user is an admin. It was working before because we only one permission but now we have more permissions. Admins have `Admin:*` (or rather `*:*`) so that should not change anything for them.

FWIW, this would be a problem for users with `Admin:Advanced` only (but I don't think we have that) and admins would need to give `Admin:ScannersResultsView`, etc. to these users.

File: src/olympia/scanners/admin.py (hunk 1 above)
Comment (willdurand) [reply src/olympia/scanners/admin.py]: We don't for this one specifically, but both use the same (private) method, which is tested _via_ the `test_list_diplay()` and `test_list_display_for_non_admins()`.

File: src/olympia/scanners/tests/test_admin.py (hunk 2 above)
Comment (willdurand) [reply src/olympia/scanners/tests/test_admin.py]: Same reason, to "fix" existing test cases.
